### PR TITLE
add NoMesh option

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**
(needed for creating a mesh without water layer)

### CODE REVIEW

File: src/input/SimModSuite.h [hunk 1]
```diff
@@ -527,6 +527,31 @@ class SimModSuite : public MeshInput {
         }
       }
     }
+    if (MeshAtt.lFaceIdNoMesh.size() > 0) {
+      for (std::list<int>::iterator it = MeshAtt.lFaceIdNoMesh.begin(); it != MeshAtt.lFaceIdNoMesh.end();
```
Comment (krenzland) [inline src/input/SimModSuite.h]: Should be `for (auto& elem : MeshAtt.lFaceIdNoMesh) {` in modern C++

File: src/input/SimModSuite.h [hunk 2]
```diff
@@ -527,6 +527,31 @@ class SimModSuite : public MeshInput {
...
+           it++) {
+        pGEntity face = GM_entityByTag(model, 2, *it);
+        if (face == NULL) {
```
Comment (krenzland) [inline src/input/SimModSuite.h]: NULL should be replaced by nullptr in modern C++

Comment (krenzland) [reply src/input/SimModSuite.h]: (Is more typesafe, as it cannot be implicitly converted to an int)

File: src/input/SimModSuite.h [hunk 3]
```diff
@@ -527,6 +527,31 @@ class SimModSuite : public MeshInput {
         }
       }
     }
+    if (MeshAtt.lFaceIdNoMesh.size() > 0) {
```
Comment (krenzland) [inline src/input/SimModSuite.h]: You can remove this check, if size is zero you just iterate over an empty list

**Review by krenzland — CHANGES_REQUESTED** [review]
Hi, great, thanks!
I've made three small comments. (Same issue is also in the other loop below)

File: src/input/SimModSuite.h (hunk 1 above)
Comment (Thomas-Ulrich) [reply src/input/SimModSuite.h]: good to know (and shorter to write), thx.

File: src/input/SimModSuite.h (hunk 2 above)
Comment (Thomas-Ulrich) [reply src/input/SimModSuite.h]: fixed.

File: src/input/SimModSuite.h (hunk 3 above)
Comment (Thomas-Ulrich) [reply src/input/SimModSuite.h]: right

**Review by krenzland — APPROVED** [review]

**Comment by krenzland** [comment]
Thanks!

**Pull request merged**